### PR TITLE
[server] Fix default permissions

### LINKED ISSUE #2083: fix pylint unused-variable warning
W0612 - Warns when a variable is defined but not used.

### PR DESCRIPTION
**Pull request by csordasmarton**
> Closes #2083 

Previously some permissions were default granted if the server was started with authentication, which means that initially, every user has the permission. With these changes all permissions are `default not granted`, which means that only users whom are explicitly given the permission have it.

By default, the `start_or_get_server` function in the test cases created a server which required authentication. With these changes  we will create 2 servers - with authentication and without it. The `add_test_package_product` function will also give permissions for the current product to some users.

### CODE REVIEW

File: web/tests/libtest/codechecker.py [hunk 1]
```diff
@@ -489,19 +505,18 @@ def start_or_get_server():
...
-            print("PostgreSQL server does not create 'Default' product...")
-            print("Creating it now!")
-            default_path = os.path.join(config_dir, 'Default')
+            default_path = os.path.join(workspace_root, 'Default')
             try:
+                print("PostgreSQL server does not create 'Default' product...")
+                print("Creating it now!")
+
                 os.makedirs(default_path)
+                add_test_package_product({'viewer_host': 'localhost',
```
Comment (gyorb) [inline web/tests/libtest/codechecker.py]: Why did you move `add_test_package_product` in the try catch block?
It will only catch OSError but `add_test_package_product` probably will not raise that type of exception. 
In `add_test_package_product` probably some connection errors will happen if the server is not started yet.

Comment (csordasmarton) [reply web/tests/libtest/codechecker.py]: Yes, it will not raise an exception, but the `os.makedirs` function will raise an exception if the directory exists so the `add_test_package_product` will not be called in this case.

I've replaced the try/except block with an if block where I check the existence of this directory 
 with the `os.path.exists` function.
